Failed to execute `make check-static`

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
Do `make check-static`.

<!-- a step by step guide for reproducing the bug. -->

### 2. What did you expect to see? (Required)
Success

### 3. What did you see instead (Required)
```
# Build from source is not recommend. See https://golangci-lint.run/usage/install/
GOBIN=/Users/xia/workspace/src/github.com/pingcap/tidb/tools/bin GO111MODULE=on go install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.47.2
GO111MODULE=on CGO_ENABLED=0 tools/bin/golangci-lint run -v $(go list ./... | sed 's|github.com/pingcap/tidb/||') --config .golangci.yml
INFO [config_reader] Used config file .golangci.yml
ERRO Can't read config: can't unmarshal config by viper: 1 error(s) decoding:

* 'Version' expected a map, got 'string'
make: *** [check-static] Error 3
```

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
```
Release Version: v9.0.0-alpha-988-gf701e5ae3d
Edition: Community
Git Commit Hash: f701e5ae3db38bb63cb230dd7c785899d13ff23c
Git Branch: master
UTC Build Time: 2025-06-24 10:28:52
GoVersion: go1.23.10
```

--- a/Makefile
+++ b/Makefile
@@ -59,6 +59,7 @@ fmt: ## Format Go code using gofmt
 	@gofmt -s -l -w -r 'interface{} -> any' $(FILES) 2>&1 | $(FAIL_ON_STDOUT)
 
 .PHONY: check-static
+check-static: ## Run static code analysis checks
 check-static: tools/bin/golangci-lint
 	GO111MODULE=on CGO_ENABLED=0 tools/bin/golangci-lint run -v $$($(PACKAGE_DIRECTORIES)) --config .golangci.yml
 
@@ -345,8 +346,8 @@ tools/bin/errdoc-gen:
 
 .PHONY: tools/bin/golangci-lint
 tools/bin/golangci-lint:
-	# Build from source is not recommend. See https://golangci-lint.run/usage/install/
-	GOBIN=$(shell pwd)/tools/bin $(GO) install github.com/golangci/golangci-lint/cmd/golangci-lint@v1.47.2
+	$(eval GOLANGCI_LINT_VERSION := $(shell grep 'github.com/golangci/golangci-lint/v2' go.mod | awk '{print $$2}'))
+	GOBIN=$(shell pwd)/tools/bin $(GO) install github.com/golangci/golangci-lint/v2/cmd/golangci-lint@$(GOLANGCI_LINT_VERSION)
 
 .PHONY: tools/bin/vfsgendev
 tools/bin/vfsgendev:

